packages/stream-tile-handler/src/tile-document-handler.ts

```diff
@@ -210,8 +217,41 @@ export class TileDocumentHandler implements StreamHandler<TileDocument> {
   async _verifySignature(
     commitData: CommitData,
     context: Context,
-    controller: string
+    controller: string,
+    streamId?: StreamID
   ): Promise<void> {
+    if (StreamUtils.isSignedCommitData(commitData)) {
```

Comment:
Can you please pull all of this new functionality out into its own helper function?  Then you'll be able to use early returns to remove indentation layers and improve readability